test/dart2_suggestors/props_and_state_companion_class_remover.suggestor_test

```diff
@@ -0,0 +1,259 @@
+PropsAndStateCompanionClassRemover
+>>> empty file (patches 0)
+<<<
+
+
+>>> no matches (patches 0)
+library foo;
+var a = 'b';
+class Foo {}
+<<<
+library foo;
+var a = 'b';
+class Foo {}
+
+
+>>> already removed (patches 0)
+@Props() class _$Foo {}
+<<<
+@Props() class _$Foo {}
+
+
+>>> @Props() public (patches 1)
+@Props() class _$Foo {}
+
+// This will be removed once the transition to Dart 2 is complete.
+class Foo extends _$Foo
+    with
+        // ignore: mixin_of_non_class, undefined_class
+        _$FooAccessorsMixin {
+  // ignore: const_initialized_with_non_constant_value, undefined_class, undefined_identifier
+  static const PropsMeta meta = _$metaForFoo;
+}
+<<<
+@Props() class _$Foo {}
+
```

Comment:
It would be worth verifying that this only removes the companion class and nothing else.
```suggestion

>>> @Props() public removes only the companion class (patches 1)
@Props() class _$Foo {}

@State() class _$FooState {}

@Component() class FooComponent {}
 // This will be removed once the transition to Dart 2 is complete.
class Foo extends _$Foo
    with
        // ignore: mixin_of_non_class, undefined_class
        _$FooAccessorsMixin {
  // ignore: const_initialized_with_non_constant_value, undefined_class, undefined_identifier
  static const PropsMeta meta = _$metaForFoo;
}
<<<
@Props() class _$Foo {}

@State() class _$FooState {}

@Component() class FooComponent {}
```